distributed/utils.py

```diff
@@ -1608,3 +1608,10 @@ def recursive_to_dict(
         return repr(obj)
     finally:
         tok.var.reset(tok)
+
+
+def is_python_shutting_down() -> bool:
+    """Is the interpreter shutting down now?"""
+    from distributed import _python_shutting_down
```

Comment:
How about a compromise?
```suggestion
    # This import must remain local for the global variable to be
    # properly evaluated
    from distributed import _python_shutting_down
```

I don't want to block this but I'm wondering if this is not identical to https://docs.python.org/3/library/sys.html#sys.is_finalizing

Reply:
I tried it and it doesn't work. It's worth adding a note in the docstring of the function though.

Reply:
Yeah, same.  I was hopeful that it would.  That would have been very clean.

Reply:
Happy to add comments, or do whatever really if anyone finds anything that works.  I'm not picky with this implementation.  I'm just trying to find a way to remove this issue.

Reply:
That particular import isn't, I don't think, the issue.  I would be happy to move that to top level if it didn't cause a cycle.

The concern that I think @crusaderky was raising was that it's strange having this code in `__init__.py`.  Or at least that's my understanding.  Please correct me if I'm wrong.

Reply:
If the function is registered at the end of ``distributed.__init__``, it works. 
If you keep the call to atexit.register in ``distributed.__init__``, but you move it just before ``from distributed.deploy import ...``, it breaks, as ``LocalCluster.__del__`` is invoked before the atexit callback.
I believe this is the same reason why ``sys.is_finalizing`` doesn't work.

This subtle behaviour should be explained in the docstring.